### PR TITLE
Padding unit test

### PR DESCRIPTION
**Pull request by egong**
### What was wrong?
padding.py needed unit tests

### How was it fixed?
Added tests to padding.py

#### Cute Animal Picture
image

> put a cute animal picture here.

### CODE REVIEW

**Comment by egong** [comment]
@pipermerriam I got the build passing so please give your feedback on the tests.  However, I did run into two env issues which I'd like some feedback.

1- I had to change my tox.ini file (which I didn't check into git).  I changed basepython = py35: python 3.5 to python3.  Otherwise I get the interpreter error.

2- Virtualenv created a whole bunch of files that I'm attaching in a pic.  Do I gitignore these files or just never check them in?  I'm looking for the best practice here.  Thanks.
screen shot 2017-08-10 at 10 30 57 am

File: tests/core/padding-utils/test_padding.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
```
Comment (gsalgado) [inline tests/core/padding-utils/test_padding.py]: Why is this needed?

File: tests/core/padding-utils/test_padding.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "value, to_size, pad_with, expected",
+    (
+        (b"", 0, b"\x00", b""),
+        (b"", 1, b"\x00", b"\x00"),
+        (b"\x00", 1, b"\x01", b"\x00"),
+        (b"\x00", 2, b"\x01", b"\x00\x01")
+    )
+)
+def test_pad_right(value, to_size, pad_with, expected):
+    assert pad_right(value, to_size, pad_with) == expected
```
Comment (gsalgado) [inline tests/core/padding-utils/test_padding.py]: Maybe add tests for pad32 and pad32r as well?

File: tests/core/padding-utils/test_padding.py (hunk 1 above)
Comment (egong) [inline tests/core/padding-utils/test_padding.py]: Assumed it was for python 2/3 compatibility because it's imported in various places across the codebase.  Should I take it out?

Comment (gsalgado) [inline tests/core/padding-utils/test_padding.py]: I hadn't noticed it in other files before, so can't tell for sure. If the travis checks pass without it I'd say remove it

**Comment by egong** [comment]
@gsalgado I've made the requested changes.  What do you think?

File: tests/core/padding-utils/test_padding.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+from evm.utils.padding import (
+    pad_left,
+    pad_right,
+    pad32,
+    pad32r
+)
+
+test_31_length_byte_string = b"\x00" * 31
+test_32_length_byte_string = b"\x00" * 32
```
Comment (gsalgado) [inline tests/core/padding-utils/test_padding.py]: These variable names are not great as they don't really make it clear they're storing the padding byte repeated N times. Also, it feels a bit weird to have variables just for two seemingly arbitrary lengths: 31 and 32. Maybe instead you could define a padding_byte variable with the _b"\x00"_ value and then in tests use that with the length multiplier you want

**Comment by gsalgado** [comment]
LGTM. Can you squash the commits before I merge?

**Comment by egong** [comment]
@gsalgado Squashed the commits

**Comment by gsalgado** [comment]
@egong care to reword that commit message? no need to mention the squashing there. https://chris.beams.io/posts/git-commit/ has some good tips on how to write good commit messages, but no need to follow them to the letter
